README.md

```diff
@@ -1,2 +1,250 @@
 # config_utilities
-Tools to make working with configs for (ros) libraries more uniform and convenient.
+Tools to make working with config structs for (ROS) C++ libraries more uniform, readable, and convenient.
+
+* **Author:** Lukas Schmid <schmluk@mavt.ethz.ch>
+* **License:** BSD-3-Clause.
+
+### Table of contents
+* [**Why config_utilities**](#Why-config_utilities)
+* [**Installation**](#Installation)
+* [**Interfaces and Tools** (The config_utilities How To...)](#Interfaces-and-Tools)
+* [**Demos**](#Demos)
+  * [config_checker](#Config-Checker)
+  * [config](#Config)
+  * [ros_param](#ROS-Params)
+  * [inheritance](#Inheritance)
+  
+# Why config_utilities
+This library was developed to provide the infrastructure to make working with config structs for object-oriented C++ libraries as simple as possible.
+Using config_utilities based configs has the following advantages:
+
+* Having all parameters in a config struct rather than with other variables makes code clearer more readable:
+  ```c++
+  if (x_ < config_.x_max) { doMagic(); }
+  ```
+* Configs can be easily checked for validity with verbose warnings to avoid runtime issues:
+  ```c++
+  MyClass::MyClass(const Config& config) : config_(config.checkValid()) {}
+  ```
+* For projects consisting of a library and a ROS-package, the configs don't have any ROS dependency and can be used in the library.
+  In the ROS-package, configs can be created from NodeHandles without additional code required:
+  ```c++
+  MyConfig c = config_utilities::getConfigFromRos<MyConfig>(nh_private);
+  ```
+* Verbose and clear printing for debugging/verification can be setup for the entire project:
+  ```c++
+  config_utilities::Settings().default_print_width = 80;
```

Comment:
Here `config_utilities::GlobalSettings().default_print_width = 80;` might be a little clearer?